### PR TITLE
Provided direct link to swift documentation

### PR DESCRIPTION
**Pull request by OneUpWallStreet**
Instead of manually searching for the documentation providing a direct link, so the reader can reach the official swift documentation.

### CODE REVIEW

File: userdocs/diagnostics/property-wrapper-requirements.md [hunk 1]
```diff
@@ -3,4 +3,6 @@ If a type is marked with the `@propertyWrapper` attribute, it must meet certain
 
 First, all property wrapper types must have a property named `wrappedValue`. This property cannot be static and must have the same access level as the property wrapper type. If the property wrapper provides a `projectedValue` property, it is subject to the same requirements.
 
-Second, none of a property wrapper's initializers may be failable. Additionally, if a property wrapper initializer has a `wrappedValue` parameter, the type of that parameter must either be the same as the type of the `wrappedValue` property or an `@autoclosure` of that type.
\ No newline at end of file
+Second, none of a property wrapper's initializers may be failable. Additionally, if a property wrapper initializer has a `wrappedValue` parameter, the type of that parameter must either be the same as the type of the `wrappedValue` property or an `@autoclosure` of that type.
+
+To understand property wrappers in more detail check out [Propeties](https://docs.swift.org/swift-book/LanguageGuide/Properties.html) section in _The Swift Programming Language_.
```
Comment (WowbaggersLiquidLunch) [inline userdocs/diagnostics/property-wrapper-requirements.md:8]: maybe link to the property wrapper section directly? https://docs.swift.org/swift-book/LanguageGuide/Properties.html#ID617

File: userdocs/diagnostics/result-builder-methods.md [hunk 2]
```diff
@@ -54,3 +54,6 @@ struct ExampleResultBuilder {
   static func buildFinalResult(_ component: Component) -> Result { ... }
 }
 ```
+
+To understand Result Builder Methods in more detail check out [Attributes](https://docs.swift.org/swift-book/ReferenceManual/Attributes.html) section in _The Swift Programming Language_.
```
Comment (WowbaggersLiquidLunch) [inline userdocs/diagnostics/result-builder-methods.md:58]: same as above, maybe link to https://docs.swift.org/swift-book/ReferenceManual/Attributes.html#ID633 directly?

**Comment by OneUpWallStreet** [comment]
@WowbaggersLiquidLunch I have made the requested changes, please let me know if there's anything else I should do 😄

**Comment by WowbaggersLiquidLunch** [comment]
Just one last nitpick: This PR adds 2 more lines but has 5 commits. Could you squash the commits into 1, so that it's cleaner after merge?

**Comment by OneUpWallStreet** [comment]
> Just one last nitpick: This PR adds 2 more lines but has 5 commits. Could you squash the commits into 1, so that it's cleaner after merge?

I don't know how to do that but I'll look into it and squash the commits, I have another PR open and by doing things here I might mess up something in that request, I could be wrong as I'm new to this and don't exactly know how everything works.  

Thank you @WowbaggersLiquidLunch for taking the time to look at my PR 😄

**Comment by WowbaggersLiquidLunch** [comment]
> I have another PR open and by doing things here I might mess up something in that request

So long as these 2 PRs are from different branches (i.e. you're using different branches for working on each PR), then you should be fine. It seems like this is the case, because this PR doesn't contain anything unrelated.

> I don't know how to do that but I'll look into it and squash the commits

In command line, I find it the easiest to use interactive rebase to squash commits. If you type in the following command:

```sh
git rebase -i HEAD~5
# '-i' tells git that you want the rebase to be interactive
# 'HEAD' is where you currently are at the git tree, and in this case your latest commit 97a362edb746ed10920dcf3388a75416bf4813cf
# 'HEAD~5' means you want to rebase the last 5 commits (including HEAD)
```

then you would a file like this opened in your default editor:

```git
pick e615d33 Provided direct link to swift documentation.
pick b464528 Added official docs link for Attributes
pick 6966ee1 added direct link to property wrappers.
pick 9da1706 added direct link to result builder methods
pick 97a362e Merge branch 'apple:main' into patch-1

# (and followed by some instructional messages from git)
```

This is the list of commits that will be rebased in chronological order. "pick" means keeping the commit unchanged. To squash a commit to its parent, replace the "pick" on its line with "squash", like this:

```git
pick e615d33 Provided direct link to swift documentation.
squash b464528 Added official docs link for Attributes
squash 6966ee1 added direct link to property wrappers.
squash 9da1706 added direct link to result builder methods
squash 97a362e Merge branch 'apple:main' into patch-1

# (and followed by some instructional messages from git)
```

Changing "pick" to "squash" for the latter 4 commits squashes each of them into their parent, resulting in only 1 commit. However, this is not done yet. You're editing a file that instruct git what it should do, so you need to save it and close it for git to use this instruction to work. 

For each resulting commit that changes, git will open a new file in your default editor, asking you what the new commit message should be. Edit the message, save, and close, and your commits are rebased.

---

Some side notes: 

I've omitted on other things you can do when rebasing, because the current focus is only squashing commits.

I've omitted on what you should do if there is a merge conflict when rebasing, because there shouldn't be any conflict for this particular rebase.

The default editor in the terminal is often vim, which new users often find difficult to use. You can set other editors such as nano, pico, or emacs, or even an external editor (that you can use outside the terminal) as the default editor.

Often it's much easier to use an app with GUI than command line for rebase. Personally, I like using Git Tower for things like this, but partially it's because I can use it for free as a student.

One more tip: I don't think this is often included in git rebase tutorials, but you can rearrange the order of your commits by rearranging the list of commits in the file that git shows you when you start an interactive rebase.

**Comment by OneUpWallStreet** [comment]
> > I have another PR open and by doing things here I might mess up something in that request
> 
> So long as these 2 PRs are from different branches (i.e. you're using different branches for working on each PR), then you should be fine. It seems like this is the case, because this PR doesn't contain anything unrelated.
> 
> > I don't know how to do that but I'll look into it and squash the commits
> 
> In command line, I find it the easiest to use interactive rebase to squash commits. If you type in the following command:
> 
> ```shell
> git rebase -i HEAD~5
> # '-i' tells git that you want the rebase to be interactive
> # 'HEAD' is where you currently are at the git tree, and in this case your latest commit 97a362edb746ed10920dcf3388a75416bf4813cf
> # 'HEAD~5' means you want to rebase the last 5 commits (including HEAD)
> ```
> 
> then you would a file like this opened in your default editor:
> 
> ```
> pick e615d33 Provided direct link to swift documentation.
> pick b464528 Added official docs link for Attributes
> pick 6966ee1 added direct link to property wrappers.
> pick 9da1706 added direct link to result builder methods
> pick 97a362e Merge branch 'apple:main' into patch-1
> 
> # (and followed by some instructional messages from git)
> ```
> 
> This is the list of commits that will be rebased in chronological order. "pick" means keeping the commit unchanged. To squash a commit to its parent, replace the "pick" on its line with "squash", like this:
> 
> ```
> pick e615d33 Provided direct link to swift documentation.
> squash b464528 Added official docs link for Attributes
> squash 6966ee1 added direct link to property wrappers.
> squash 9da1706 added direct link to result builder methods
> squash 97a362e Merge branch 'apple:main' into patch-1
> 
> # (and followed by some instructional messages from git)
> ```
> 
> Changing "pick" to "squash" for the latter 4 commits squashes each of them into their parent, resulting in only 1 commit. However, this is not done yet. You're editing a file that instruct git what it should do, so you need to save it and close it for git to use this instruction to work.
> 
> For each resulting commit that changes, git will open a new file in your default editor, asking you what the new commit message should be. Edit the message, save, and close, and your commits are rebased.
> 
> Some side notes:
> 
> I've omitted on other things you can do when rebasing, because the current focus is only squashing commits.
> 
> I've omitted on what you should do if there is a merge conflict when rebasing, because there shouldn't be any conflict for this particular rebase.
> 
> The default editor in the terminal is often vim, which new users often find difficult to use. You can set other editors such as nano, pico, or emacs, or even an external editor (that you can use outside the terminal) as the default editor.
> 
> Often it's much easier to use an app with GUI than command line for rebase. Personally, I like using Git Tower for things like this, but partially it's because I can use it for free as a student.
> 
> One more tip: I don't think this is often included in git rebase tutorials, but you can rearrange the order of your commits by rearranging the list of commits in the file that git shows you when you start an interactive rebase.

Hey @WowbaggersLiquidLunch thank you so much for writing this detailed explanation, I went through it and followed the steps but it looks like I messed up somewhere and now its telling me that there is some merge conflict. I tried to fix it but I'm not able to. 
Will it be okay if I just close this pull request and open a new one with those changes in a single commit? 😄

**Comment by WowbaggersLiquidLunch** [comment]
You can abort a rebase with `git rebase --abort`. 

> Will it be okay if I just close this pull request and open a new one with those changes in a single commit?

Yes you can do that.

**Pull request closed without merging**